spring-boot-project/spring-boot-actuator-autoconfigure/src/test/java/org/springframework/boot/actuate/autoconfigure/metrics/PropertiesMeterFilterTests.java

```diff
@@ -296,6 +296,110 @@ public void configureWhenSlaLongShouldOnlyApplyToTimerAndDistributionSummary() {
 				.getSlaBoundaries()).isNullOrEmpty();
 	}
 
+	@Test
+	public void configureWhenHasMinimumExpectedValueShouldSetMinimumExpectedToValue() {
+		PropertiesMeterFilter filter = new PropertiesMeterFilter(
+				createProperties("distribution.minimum-expected-value.[spring.boot]=10"));
+		assertThat(filter.configure(createMeterId("spring.boot"),
+				DistributionStatisticConfig.DEFAULT).getMinimumExpectedValue())
+						.isEqualTo(10000000);
+	}
+
+	@Test
+	public void configureWhenHasHigherMinimumExpectedValueShouldSetMinimumExpectedValueToValue() {
+		PropertiesMeterFilter filter = new PropertiesMeterFilter(
+				createProperties("distribution.minimum-expected-value.spring=10"));
+		assertThat(filter.configure(createMeterId("spring.boot"),
+				DistributionStatisticConfig.DEFAULT).getMinimumExpectedValue())
+						.isEqualTo(10000000);
+	}
+
+	@Test
+	public void configureWhenHasHigherMinimumExpectedValueAndLowerShouldSetMinimumExpectedValueToHigher() {
+		PropertiesMeterFilter filter = new PropertiesMeterFilter(
+				createProperties("distribution.minimum-expected-value.spring=10",
+						"distribution.minimum-expected-value.[spring.boot]=50"));
+		assertThat(filter.configure(createMeterId("spring.boot"),
+				DistributionStatisticConfig.DEFAULT).getMinimumExpectedValue())
+						.isEqualTo(50000000);
+	}
+
+	@Test
+	public void configureWhenAllMinimumExpectedValueSetShouldSetMinimumExpectedValueToValue() {
+		PropertiesMeterFilter filter = new PropertiesMeterFilter(
+				createProperties("distribution.minimum-expected-value.all=10"));
+		assertThat(filter.configure(createMeterId("spring.boot"),
+				DistributionStatisticConfig.DEFAULT).getMinimumExpectedValue())
+						.isEqualTo(10000000);
+	}
+
+	@Test
+	public void configureWhenMinimumExpectedValueDurationShouldOnlyApplyToTimer() {
+		PropertiesMeterFilter filter = new PropertiesMeterFilter(
+				createProperties("distribution.minimum-expected-value.all=10ms"));
+		Meter.Id timer = createMeterId("spring.boot", Meter.Type.TIMER);
+		Meter.Id summary = createMeterId("spring.boot", Meter.Type.DISTRIBUTION_SUMMARY);
+		Meter.Id counter = createMeterId("spring.boot", Meter.Type.COUNTER);
+		assertThat(filter.configure(timer, DistributionStatisticConfig.DEFAULT)
+				.getMinimumExpectedValue()).isEqualTo(10000000L);
+		assertThat(filter.configure(summary, DistributionStatisticConfig.DEFAULT)
+				.getMinimumExpectedValue()).isEqualTo(1L);
+		assertThat(filter.configure(counter, DistributionStatisticConfig.DEFAULT)
+				.getMinimumExpectedValue()).isEqualTo(1L);
+	}
+
+	@Test
+	public void configureWhenHasMaximumExpectedValueShouldSetMaximumExpectedToValue() {
+		PropertiesMeterFilter filter = new PropertiesMeterFilter(createProperties(
+				"distribution.maximum-expected-value.[spring.boot]=5000"));
+		assertThat(filter.configure(createMeterId("spring.boot"),
+				DistributionStatisticConfig.DEFAULT).getMaximumExpectedValue())
+						.isEqualTo(5000000000L);
```

Comment:
For those of us that aren't used to seeing the difference in significant digits between nanoseconds and milliseconds, it might be more clear to write `Duration.ofMillis(5000).toNanos()`